Tests/Remove-GitItem.Tests.ps1

```diff
@@ -0,0 +1,197 @@
+        It ('should not stage {0}' -f $pathItem) {
+            Get-GitRepositoryStatus -RepoRoot $repoRoot -Path $pathItem | 
+                Where-Object { $_.IsStaged } |
+                Measure-Object | 
+                Select-Object -ExpandProperty 'Count' | 
+                Should Be 0
+        }
+    }
+}
+
+function ThenFileShouldBeDeleted
+{
+    param(
+        [string[]]
+        $Path
+    )
+
+    foreach( $pathItem in $Path )
+    {
+        It ('should have deleted {0}' -f $pathItem) {
+            Test-Path -Path (join-Path -Path $repoRoot -ChildPath $pathItem) | Should -be $false
+        }
+    }
+}
+
+function ThenNoErrorShouldBeThrown
+{
+    $Global:Error | Should -BeNullOrEmpty
+}
+function ThenErrorShouldBeThrown
+{
+    param(
+        [String]
+        $ExpectedError
+    )
+    It ('Should throw an error matching ''{0}''' -f $ExpectedError){
+        $Global:Error | Where-Object { $_ -match $ExpectedError } | Should -Not -BeNullOrEmpty
+    }
+}
+
+Describe 'Remove-GitItem.When File is moved from git repository correctly' {
+    GivenRepositoryWithFile -Name 'foo.bar'
+    GivenFileToStage -Name 'foo.bar'
+    WhenFileIsStaged
+    ThenFileShouldBeStaged -Path 'foo.bar'
+    ThenFileShouldBeDeleted -Path 'foo.bar'
+    ThenNoErrorShouldBeThrown
+}
+
+Describe 'Remove-GitItem.When multiple Files are moved from git repository correctly' {
+    GivenRepositoryWithFile -Name 'foo.bar', 'bar.fooo'
+    GivenFileToStage  -Name 'foo.bar', 'bar.fooo'
+    WhenFileIsStaged
+    ThenFileShouldBeStaged -Path 'foo.bar', 'bar.fooo'
+    ThenFileShouldBeDeleted -Path 'foo.bar', 'bar.fooo'
+    ThenNoErrorShouldBeThrown
+}
+
+Describe 'Remove-GitItem.When multiple Files are moved from git repository correctly via the pipeline' {
+    GivenRepositoryWithFile -Name 'foo.bar', 'bar.fooo'
+    GivenFileToStage  -Name 'foo.bar', 'bar.fooo'
+    WhenFileIsStagedByPipeline
+    ThenFileShouldBeStaged -Path 'foo.bar', 'bar.fooo'
+    ThenFileShouldBeDeleted -Path 'foo.bar', 'bar.fooo'
+    ThenNoErrorShouldBeThrown
+}
+
+Describe 'Remove-GitItem.When File is moved from git repository correctly via the pipeline' {
+    GivenRepositoryWithFile -Name 'foo.bar'
+    GivenFileToStage -Name 'foo.bar'
+    WhenFileIsStagedByPipeline
+    ThenFileShouldBeStaged -Path 'foo.bar'
+    ThenFileShouldBeDeleted -Path 'foo.bar'
+    ThenNoErrorShouldBeThrown
+}
+
+Describe 'Remove-GitItem.When File is already removed from git repository correctly' {
+    GivenRepositoryWithFile -Name 'foo.bar'
+    GivenFileIsDeleted -Name 'foo.bar'
+    GivenFileToStage  -Name 'foo.bar'
+    WhenFileIsStaged
+    ThenFileShouldBeStaged -Path 'foo.bar'
+    ThenFileShouldBeDeleted -Path 'foo.bar'
+    ThenNoErrorShouldBeThrown
+}
+
+Describe 'Remove-GitItem.When File doesnt exist in the repository' {
+    GivenRepositoryWithFile -Name 'a.file'
+    GivenFileToStage  -Name 'different.file'
+    WhenFileIsStaged
+    ThenFileShouldNotBeStaged -Path 'different.file'

```

Comment:
Add `ThenNoErrorShouldBeThrown` assertion.

Reply:
Done

Reply:
Done, added release notes and build is fixed